smarts/sstudio/types.py

```diff
@@ -565,21 +542,27 @@ def pick_remaining_shape_after_split(geometry_collection, length, lane):
 
             min_cut = min(lane_offset, lane_length)
             # Second cut takes into account shortening of geometry by `min_cut`.
-            max_cut = min(min_cut + geom_length, lane_length - min_cut)
+            max_cut = min(min_cut + geom_length, lane_length)
```

Comment:
I assume that max_cut should always be greater than min_cut, `lane_length - min_cu`t would make max_cut less than min_cut.

Reply:
I think this means we need to have a test to make sure that extremely short lanes and extremely long lane geometry do not cause errors.

It is correct to limit the `max_cut` to the `lane_length`, if we want to push back off the end of the lane it should be calculated in the `min_cut`.

Reply:
Now it makes more sense. I think Aurora is right, the max_cut should less than the length.